src/main/scala/nl/knaw/dans/easy/archivebag/EasyArchiveBag.scala

```diff
@@ -16,44 +16,86 @@
 
 import java.io._
-import java.net.{URI, URL}
-import java.nio.file.{Files, Paths}
+import java.net.URI
+import java.nio.file.{Files, Path, Paths}
 import java.util.zip.{ZipEntry, ZipOutputStream}
 
 import org.apache.commons.codec.digest.DigestUtils
-import org.apache.commons.codec.net.URLCodec
-import org.apache.commons.io.IOUtils
+import org.apache.commons.io.{FileUtils, IOUtils}
 import org.apache.http.auth.{AuthScope, UsernamePasswordCredentials}
-import org.apache.http.client.methods.{CloseableHttpResponse, HttpGet, HttpPost, HttpPut}
+import org.apache.http.client.methods.{CloseableHttpResponse, HttpGet, HttpPut}
 import org.apache.http.entity.{ContentType, FileEntity}
 import org.apache.http.impl.client.{BasicCredentialsProvider, CloseableHttpClient, HttpClients}
 import org.slf4j.LoggerFactory
 
-import scala.util.{Failure, Try}
-import scala.xml._
+import scala.util.{Success, Try}
 
-object EasyArchiveBag {
+object EasyArchiveBag extends Bagit4FacadeComponent {
   val log = LoggerFactory.getLogger(getClass)
+  type BagId = String
+  val bagFacade = new Bagit4Facade()
 
   def main(args: Array[String]) {
     implicit val settings = CommandLineOptions.parse(args)
     run.get
   }
 
   def run(implicit ps: Parameters): Try[URI] = Try {
+    for {
+      optVersionOfId <- getIsVersionOf(ps.bagDir.toPath)
+      optRefBagsTxt <- optVersionOfId.map(getBagSequence).getOrElse(Success(None))
+      _ <- optRefBagsTxt.map(writeRefBagsTxt(ps.bagDir.toPath)).getOrElse(Success(()))
+    } yield ()
+
     val zippedBag = generateUncreatedTempFile()
     zipDir(ps.bagDir, zippedBag)
     val response = putFile(zippedBag)
     response.getStatusLine.getStatusCode match {
       case 201 =>
         val location = new URI(response.getFirstHeader("Location").getValue)
         log.info(s"Bag archival location created at: $location")
+        addBagToIndex(ps.uuid.toString).recover {
+          case t => log.warn(s"BAG ${ps.uuid} NOT ADDED TO BAG-INDEX. SUBSEQUENT REVISIONS WILL NOT BE PRUNED", t)
+        }
         location
       case _ =>
         throw new RuntimeException(s"Bag archiving failed: ${response.getStatusLine}")
     }
   }
 
+  private def addBagToIndex(bagId: BagId)(implicit ps: Parameters): Try[Unit] = Try {
+    val http = createHttpClient(ps.bagIndexService.getHost, ps.bagIndexService.getPort, "", "")
+    val put = new HttpPut(ps.bagIndexService.resolve(s"bags/$bagId").toASCIIString)
+    log.info(s"Adding new bag to bag index with request: ${put.toString}")
+    val response = http.execute(put)
+    if(response.getStatusLine.getStatusCode != 201) throw new IllegalStateException("Error trying to add bag to index")
+  }
+
+  private def getIsVersionOf(bagDir: Path): Try[Option[BagId]] = {
+    for {
+      info <- bagFacade.getBagInfo(bagDir)
+      versionOf <- Try { info.get(IS_VERSION_OF_KEY) }
+    } yield versionOf
+  }
+
+
+  private def getBagSequence(bagId: BagId)(implicit ps: Parameters): Try[Option[String]] = {
```

Comment:
would you mind documenting this function, especially what the resulting `String` contains exactly? I had to look for a couple of minutes to find that this is a newline separated list of bagIds in the same sequence... A little documentation would help here!